.gitattributes

```
@@ -0,0 +1,2 @@
+# Set the default behavior, in case people don't have core.autocrlf set.
+* text=auto eol=lf
```

I must admit I don't full know what this gitattributes will do; I worry that it'll apply eol transforms against binary files or some other unexpected thing. Other projects seem to list out the whole list of file extensions manually, e.g. Drupal: https://github.com/alexkaratarakis/gitattributes/blob/master/Drupal.gitattributes - I assume there must be a reason for this?

Something like this:

```suggestion
* text=auto

*.js text eol=lf
*.jsx text eol=lf
*.ts text eol=lf
*.tsx text eol=lf
*.json text eol=lf
*.md text eol=lf
*.html text eol=lf
*.css text eol=lf
*.sh text eol=lf
*.yml text eol=lf
```

---

Its all a good excuse to learn new stuff!

The drupal one is quite verbose ain't it? I read through the doc it links to (https://mirrors.edge.kernel.org/pub/software/scm/git/docs/gitattributes.html) and there are some bits that jumped out which seem to valid the initial logic:

"Git recognizes files encoded in ASCII or one of its supersets (e.g. UTF-8, ISO-8859-1, …) as text files. Files encoded in certain other encodings (e.g. UTF-16) are interpreted as binary and consequently built-in Git text processing tools (e.g. git diff) as well as most Git web front ends do not visualize the contents of these files by default."

However now you've written the above might as well roll with that as its explicit. Does mean we might need to add to it in the future if new file type is added but cross that bridge when we come to it I guess.

---

:+1: Thanks Pete